pkg/asset/parents.go

```diff
@@ -0,0 +1,15 @@
+package asset
+
+import (
+	"reflect"
+)
+
+// Parents is the collection of states for the assets upon which another asset
+// is directly dependent.
+type Parents map[reflect.Type]*State
+
+// Get gets the state of the specified asset.
+func (p Parents) Get(asset Asset) (*State, bool) {
```

Comment:
Why is there a distinction between `Asset` and `State`? Is it possible for each asset to implement its own special logic and the `Asset` interface, allowing us to drop `State`? I had imagined this method would look as follows:

```go
func (p Parents) Get(asset &Asset) { // this isn't quite right because Asset is an interface
  *asset, ok := p[reflect.TypeOf(asset)]
  if !ok {
    panic("Asset %s was requested, but was not available", asset.Name())
  }
}
```

We could then pass in the subset of assets that have been declared by each assets' `Dependencies()` method to ensure that every asset is properly declaring its dependencies.